src/routes/download/drs.service.ts

```diff
@@ -0,0 +1,70 @@
+import urlJoin from 'url-join';
+import fetch from 'node-fetch';
+
+import { AppConfig } from '@/config/global';
+import logger from '@/logger';
+
+// types from https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/#_accessmethod
+// will only be using https
+enum AccessType {
+  HTTPS = 'https',
+  S3 = 's3',
+  GS = 'gs',
+  FTP = 'ftp',
+  FILE = 'file',
+  GSIFTP = 'gsiftp',
+  GLOBUS = 'globus',
+  HTSGET = 'htsget',
+}
+
+interface AccessMethod {
+  type: AccessType;
+  access_url: { url: string };
+  region: string;
+}
+
+export interface DRSMetada {
+  name: string;
+  accessUrls: AccessMethod[];
+}
+
+export const parseDRSPath = (filepath: string, config: AppConfig) => {
+  const toUri = new URL(filepath);
+  const domain = toUri.hostname;
+  const objectId = toUri.pathname;
+  return urlJoin(`${config.drs.protocol}://${domain}`, config.drs.objectPath, objectId);
+};
+
+const getAccessUrl: (accessMethods: AccessMethod[]) => AccessMethod[] = (accessMethods) => {
+  // return all https access urls. at least one of this type is guaranteed per drs path
+  return accessMethods.filter((method) => method.type === AccessType.HTTPS);
+};
+
+export const fetchDRSMetadata = async (drsUri: string) => {
+  const response = await fetch(drsUri)
+    .then((res) => {
+      if (res.status !== 200) {
+        throw new Error(`Response from drs url failed with ${res.status} error.`);
+      }
+      return res.json();
+    })
+    .then((json) => {
+      return {
+        accessUrls: getAccessUrl(json.access_methods),
+        name: json.name,
+      } as DRSMetada;
```

Comment:
since all `getAccessUrl` does is check for `HTTPS` in the list, is this actually just doing:

`if (json.access_methods.find(AccessMethods.HTTPS)) {return json.name}; else { // throw error or return null }` 
?